### ISSUE TITLE
FA100 doesn't catch using modern annotation syntax when target version is 3.9

### ISSUE THREAD

**Issue by akx** [charliermarsh/ruff#4553]
As I'm reading it, FA100 should add a `from __future__ import annotations` import when a file already uses modern annotation syntax and the configuration targets Python 3.9, but it doesn't:

Refs https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/10580

```
$ cat a.py
def foo(x: str | None):
	pass
$ ruff --version
ruff 0.0.269
$ ruff --target-version=py39 --select=FA100 --isolated a.py
$ ~/.pyenv/versions/3.9.16/bin/python3.9 a.py
Traceback (most recent call last):
  File "a.py", line 1, in <module>
    def foo(x: str | None):
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

**Comment by charliermarsh** [comment]
I believe the intent of `FA100` is that it should trigger when it finds a type annotation you _could_ rewrite _if_ you enabled future annotations. So, e.g., if you had `x: Optional[str]`, _then_ it might trigger. But I don't believe it's intended to trigger in the case you outlined above. Am I misunderstanding? Or are the docs misrepresenting the rule in some way?

**Comment by akx** [comment]
Well, in that case another FA* rule that would address this would probably be useful?

I.e. if the target Python version is <= 3.9, then any file using modern type annotations would also have to have a `from __future__ import annotations` . (PyCharm's "version compatibility" lint does this, FWIW.)

**Comment by charliermarsh** [comment]
I think that might be `FA102`: https://pypi.org/project/flake8-future-annotations/

**Comment by akx** [comment]
@charliermarsh Ah, yeah, thanks! Someone oughta get implementing FA102 in Ruff then, I guess 😂 

Closing this, I guess the rest of FA* is tracked in https://github.com/charliermarsh/ruff/issues/3072

**Comment by charliermarsh** [comment]
Lol yes :joy:
